PrivateHeaders/APPX/ZIP.h

```diff
@@ -371,6 +373,55 @@ namespace appx {
     }
 
     template <typename TSink>
+    ZIPFileEntry
+    WriteAppxBundleManifestZIPFileEntry(TSink &sink, off_t offset,
+                                        const std::string &inputFileName,
+                                        const std::string &archiveFileName,
+                                        const std::vector<ZIPFileEntry> &otherEntries)
+    {
+        std::ifstream manifestInput(inputFileName);
+        std::string manifestText((std::istreambuf_iterator<char>(manifestInput)),
+                                  std::istreambuf_iterator<char>());
+
+       std::cout << "\n\n ABOUT TO GO OVER OTHERENTRIES" << "\n\n";
+       std::cout << "\n\n" << manifestText << "\n\n";
+
+        for (const ZIPFileEntry &entry : otherEntries) {
+            std::string fileName = entry.fileName;
+            std::string offsetTemplateName = entry.fileName + "-offset";
+            std::cout << "\n\n" << offsetTemplateName;
+            size_t pos = 0;
+            size_t entryDataOffset = entry.fileRecordHeaderOffset + entry.FileRecordHeaderSize();
+            while ((pos = manifestText.find(offsetTemplateName, pos)) != std::string::npos) {
+                std::cout << "replacing something" << std::endl;
+                manifestText.replace(pos, offsetTemplateName.length(), std::to_string(entryDataOffset));
+            }
```

Comment:
Hmm, so we're replacing parts of the input AppxBundleManifest.xml file? This should be document somewhere; it's very non-obvious.